native/cocos/3d/assets/Mesh.cpp

```diff
@@ -356,6 +427,11 @@ void Mesh::initialize() {
         if (_struct.morph.has_value()) {
             morphRendering = createMorphRendering(this, gfxDevice);
         }
+
+        _isMeshDataUploaded = true;
+        if (!_allowReadWriteData) {
+            releaseData();
+        }
```

Comment:
It could be.
But I like the current code more than that. Because while reading the code, I could simply know the logic (condition) before doing the real `releaseData` operation. E.g.

```c++
_isMeshDataUploaded = true;
  if (!_allowReadWriteData) { // This `if` code say that: I'm a mesh doesn't allow read/write operation, so please releaseData now
      releaseData();
  }
```

```c++
_isMeshDataUploaded = true;
releaseData(); // Could data be released, may not! Because it depends on the logic inside releaseData. While reading code here without knowing the implementation of releaseData function, code reader may assume that it just doing the actual `release` operation.
```

Reply:
You can rename to `releaseDataIfPossible`